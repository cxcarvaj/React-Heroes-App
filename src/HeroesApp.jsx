export const HeroesApp = () => {
    return(
        <>
            <h1>Heroes App</h1>
        </>
    );
};