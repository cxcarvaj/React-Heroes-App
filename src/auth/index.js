export * from './context';
export * from './pages';