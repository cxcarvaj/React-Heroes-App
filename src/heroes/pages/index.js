export * from './MarvelPage';
export * from './DcPage';
export * from './SearchPage';
export * from './HeroPage';