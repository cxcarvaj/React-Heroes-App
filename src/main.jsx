import React from 'react';
import ReactDOM from 'react-dom/client';
import { BrowserRouter } from 'react-router-dom';

import { HeroesApp } from './HeroesApp';
import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <BrowserRouter>
    <React.StrictMode>
      <HeroesApp />
    </React.StrictMode>
  </BrowserRouter>
);
