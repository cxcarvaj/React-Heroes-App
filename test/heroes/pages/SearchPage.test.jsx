import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { SearchPage } from '../../../src/heroes';


const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'), 
    useNavigate: () => mockedUseNavigate,
}));


describe('Test in <SearchPage />', () => {

    beforeEach(() => {
        jest.clearAllMocks();
    });

    test('should show correctly with default values', () => {

        const { container } = render(
            <MemoryRouter>
                <SearchPage />
            </MemoryRouter>
        );
        expect(container).toMatchSnapshot();

    });

    test('should show Batman information and the input with the value of the queryString', () => {

        render(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <SearchPage />
            </MemoryRouter>
        );

        const input = screen.getByRole('textbox');

        expect( input.value ).toBe('batman');

        const img = screen.getByRole('img');

        expect( img.src ).toContain('/assets/heroes/dc-batman.jpg');

        const alertSearchHero = screen.getByLabelText('alert-search-hero');

        const alertNoHero = screen.getByLabelText('alert-no-hero');


        expect( alertSearchHero.style.display ).toBe('none');
        expect( alertNoHero.style.display ).toBe('none');

    });

    test('should show an error if the hero does not exist', () => {

        render(
            <MemoryRouter initialEntries={['/search?q=batman123']}>
                <SearchPage />
            </MemoryRouter>
        );

        const alertNoHero = screen.getByLabelText('alert-no-hero');

        expect( alertNoHero.style.display ).not.toBe('none');

    });

    test('should call the push of history', () => {
        render(
            <MemoryRouter initialEntries={['/search?q=batman123']}>
                <SearchPage />
            </MemoryRouter>
        );

        const input = screen.getByRole('textbox');
        
        fireEvent.change( input, { target: { name: 'searchText', value: 'flash' } } );

        const form = screen.getByRole('form');

        fireEvent.submit( form );

        expect( mockedUseNavigate ).toHaveBeenCalled();
        expect( mockedUseNavigate ).toHaveBeenCalledWith('?q=flash');

    });

});