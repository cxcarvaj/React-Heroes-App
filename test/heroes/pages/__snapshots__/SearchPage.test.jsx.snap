// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <SearchPage /> should show correctly with default values 1`] = `
<div>
  <div>
    <h1>
      Search
    </h1>
    <hr />
    <div
      class="row"
    >
      <div
        class="col-5"
      >
        <h4>
          Searching
        </h4>
        <hr />
        <form
          aria-label="form"
        >
          <input
            autocomplete="off"
            class="form-control"
            name="searchText"
            placeholder="Search a hero"
            type="text"
            value=""
          />
          <button
            class="btn btn-outline-primary mt-1"
          >
            Search
          </button>
        </form>
      </div>
      <div
        class="col-7"
      >
        <h4>
          Results
        </h4>
        <hr />
        <div
          aria-label="alert-search-hero"
          class="alert alert-primary animate__animated animate__fadeIn animate__fast"
        >
          Search a hero
        </div>
        <div
          aria-label="alert-no-hero"
          class="alert alert-danger animate__animated animate__fadeIn animate__fast"
          style="display: none;"
        >
          No hero with 
          <b />
        </div>
      </div>
    </div>
  </div>
</div>
`;
